### PR TITLE
feat: Callback method for hotswap solutions

### PR DESCRIPTION
**Pull request by Artur-**
Hotswapper.onHotswap can be called by a list of changed classes (as strings) by any hotswap solution and Hilla will react to the changed classes in an appropriate way, for instance by re-generating the endpoint metadata and TypeScript. The hotswap solution is not supposed to do any pre-filtering of the changed classes, this is done by the method itself.

### CODE REVIEW

**Comment by Artur-** [comment]
Based on #1172  and #1163

File: packages/java/endpoint/src/main/java/dev/hilla/EndpointCodeGenerator.java [hunk 1]
```diff
@@ -36,8 +37,12 @@ public class EndpointCodeGenerator {
     private final EndpointController endpointController;
     private final VaadinContext context;
     private Path buildDirectory;
+
     private ApplicationConfiguration configuration;
     private String nodeExecutable;
+    private Set<String> classesUsedInOpenApi = null;
+
+    private static EndpointCodeGenerator instance = null;
```
Comment (cromoteca) [inline packages/java/endpoint/src/main/java/dev/hilla/EndpointCodeGenerator.java:45]: Where's its value initialized?

Comment (Artur-) [reply packages/java/endpoint/src/main/java/dev/hilla/EndpointCodeGenerator.java:45]: Feels like something has gone missing during the rebase

Comment (Artur-) [reply packages/java/endpoint/src/main/java/dev/hilla/EndpointCodeGenerator.java:45]: Added

**Review by cromoteca — APPROVED** [review]

**Pull request merged**